apps/studio/components/ui/FileExplorerAndEditor/FileExplorerAndEditor.tsx

```diff
@@ -0,0 +1,191 @@
+    const updatedFiles = files.map((f) => ({ ...f, selected: false }))
+    onFilesChange([
+      ...updatedFiles,
+      {
+        id: newId,
+        name: `file${newId}.ts`,
+        content: '',
+        selected: true,
+      },
+    ])
+  }
+
+  const handleFileNameChange = (id: number, newName: string) => {
+    if (!newName.trim()) return // Don't allow empty names
+    const updatedFiles = files.map((file) => (file.id === id ? { ...file, name: newName } : file))
+    onFilesChange(updatedFiles)
+  }
+
+  const handleDoubleClick = (id: number, currentName: string) => {
+    const newName = prompt('Enter new file name:', currentName)
+    if (newName && newName !== currentName) {
+      handleFileNameChange(id, newName)
+    }
+  }
+
+  const handleFileSelect = (id: number) => {
+    const updatedFiles = files.map((file) => ({
+      ...file,
+      selected: file.id === id,
+    }))
+    onFilesChange(updatedFiles)
+  }
+
+  const treeData = {
+    name: '',
+    children: files.map((file) => ({
+      id: file.id.toString(),
+      name: file.name,
+      metadata: {
+        isEditing: false,
+        originalId: file.id,
+      },
+    })),
+  }
+
+  return (
+    <div className="flex-1 overflow-hidden flex h-full">
+      <div className="w-64 border-r bg-surface-200 flex flex-col">
+        <div className="py-4 px-6 border-b flex items-center justify-between">
+          <h3 className="text-sm font-medium">Files</h3>
+          <Button size="tiny" type="default" icon={<Plus size={14} />} onClick={addNewFile}>
+            Add File
+          </Button>
+        </div>
+        <div className="flex-1 overflow-y-auto p-2">
+          <TreeView
+            data={flattenTree(treeData)}
+            aria-label="files tree"
+            nodeRenderer={({ element, isBranch, isExpanded, getNodeProps, level }) => {
+              const nodeProps = getNodeProps()
+              const originalId =
+                typeof element.metadata?.originalId === 'number'
+                  ? element.metadata.originalId
+                  : null
+
+              return (
+                <div
+                  onDoubleClick={(e) => {
+                    e.stopPropagation()
+                    if (originalId !== null) handleDoubleClick(originalId, element.name)
+                  }}
+                >
+                  <TreeViewItem
+                    {...nodeProps}
+                    isExpanded={isExpanded}
+                    isBranch={isBranch}
+                    isSelected={files.find((f) => f.id === originalId)?.selected}
+                    level={level}
+                    xPadding={16}
+                    name={element.name}
+                    icon={<File size={14} className="text-foreground-light" />}
+                    isEditing={Boolean(element.metadata?.isEditing)}
+                    onEditSubmit={(value) => {
+                      if (originalId !== null) handleFileNameChange(originalId, value)
```

Comment:
Could we use a context menu to select and rename? Similar to SQL Editor for consistency (Clicking to rename was not intuitive)
there's also no way to delete the file atm i think